app/controllers/api/v1/base_controller.rb

```diff
@@ -240,8 +259,11 @@ def key_transform_header
       private
 
       def authenticate_user_token!
-        authenticate_with_http_token do |token, _options|
-          return User.includes(:language).find_by_auth_token(token)
+        authenticate_with_http_token do |auth_token, _options|
+          token = Token.includes(:user).find_by(token: auth_token)
```

Comment:
Should return early if `auth_token` is blank. I think `authenticate_with_http_token` always yields to the block and its unnecessary to do a database query then.